lumicks/pylake/kymotracker/detail/geometry_2d.py

```diff
@@ -0,0 +1,129 @@
+from scipy.ndimage import gaussian_filter
+from .linalg_2d import eigenvalues_2d_symmetric, eigenvector_2d_symmetric
+import numpy as np
+
+
+def find_subpixel_location(gx, gy, gxx, gxy, gyy, nx, ny):
+    """This function determines the subpixel location at which the second derivative attains its minimum.
+    It works by performing a Taylor expansion in the direction of the gradient (perpendicular to the line)
+    and then analytically solving for the first derivative. In 1D:
+      f(x) = r + r'x + r''x^2 / 2
+      f'(x) = r' + r''x = 0
+    ==> x = - r'/r''
+
+    Parameters
+    ----------
+    gx: array_like
+        First order Gaussian derivative in x direction.
+    gy: array_like
+        First order Gaussian derivative in y direction.
+    gxx: array_like
+        Second order Gaussian derivative in x direction.
+    gyy: array_like
+        Second order Gaussian derivative in y direction.
+    gxy: array_like
+        Gaussian derivative w.r.t. x and y.
+    nx: array_like
+        Image containing x component of the vector normal to the line.
+    ny: array_like
+        Image containing y component of the vector normal to the line.
+
+    Returns
+    -------
+    px: array_like
+        Image containing the x coordinate of the subpixel location of the ridge.
+    px: array_like
+        Image containing the x coordinate of the subpixel location of the ridge.
+    image: array_like
+        Mask whether the optimum is inside or outside of the pixel.
+    """
+
+    # Evaluate the subpixel location of the line
+    denominator = (gxx * nx * nx + 2.0 * gxy * nx * ny + gyy * ny * ny)
```

Comment:
Yes, it was indeed a cut/paste leftover. And it was actually superfluous, since that denominator is exactly equal to the largest eigenvalue (which we already computed). So I pass the eigenvalue through now instead.